internal/dataplane/kong_client.go

```diff
@@ -645,9 +645,9 @@ func UniqueObjects(reportedObjects []client.Object, resourceFailures []failures.
 	})
 }
 
-// updateKubernetesObjectReportFilter overrides the internal object set with
+// UpdateKubernetesObjectReportFilter overrides the internal object set with
 // a new provided set.
-func (c *KongClient) updateKubernetesObjectReportFilter(set k8sobj.ConfigurationStatusSet) {
+func (c *KongClient) UpdateKubernetesObjectReportFilter(set k8sobj.ConfigurationStatusSet) {
```

Comment:
I think that making this public isn't really required as it's used only internally.